### PR TITLE
[BACKLOG-40063] -  VFS Connection Manager : Allow permission to be applied on the VFS connections created from PUC

### PR DESCRIPTION
**Pull request by soagarwal1**
Part of PR group

https://github.com/pentaho/pdi-plugins-ee/pull/1050
https://github.com/pentaho/pentaho-kettle/pull/9283
https://github.com/pentaho/big-data-plugin/pull/2523

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1m 26s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
:ok_hand: All tests passed!
 
Tests run: **120**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/metastore/job/PR-89/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
**👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.**
<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1m 51s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
:ok_hand: All tests passed!
 
Tests run: **121**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/metastore/job/PR-89/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: src/test/java/org/pentaho/metastore/test/MetaStoreFactoryTest.java [hunk 1]
```diff
@@ -558,4 +559,16 @@ private List<DimensionAttribute> generateAttributes() {
     return list;
   }
 
+  @Test
```
Comment (dcleao) [inline src/test/java/org/pentaho/metastore/test/MetaStoreFactoryTest.java:562]: Please, use instead `@Test(expected = MetaStoreException.class)`

File: src/test/java/org/pentaho/metastore/test/MetaStoreFactoryTest.java [hunk 2]
```diff
@@ -558,4 +559,16 @@ private List<DimensionAttribute> generateAttributes() {
...
+  public void testGetterNotExists() {
```
Comment (dcleao) [inline src/test/java/org/pentaho/metastore/test/MetaStoreFactoryTest.java:563]: AFAICT, while valuable to test previously existing behavior, this test does not address the new feature being added: that of being able to have fields and getters/setters from base classes. It would be good to add more tests covering the new feature. Can be on a separate PR, if there is need to merge this sooner.

**Review by dcleao — CHANGES_REQUESTED** [review]

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :warning: Build finished in 1m 40s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.metastore.test.MetaStoreFactoryTest.testGetterNotExists</b> (click to expand)</summary><p>

```
Unable to find getter for attribute field : name. Expected 'getName'
```

</p></details>

  
Tests run: **121**, Failures: **1**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/metastore/job/PR-89/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: src/test/java/org/pentaho/metastore/test/MetaStoreFactoryTest.java (hunk 1 above)
Comment (soagarwal1) [reply src/test/java/org/pentaho/metastore/test/MetaStoreFactoryTest.java:562]: Making this change results in failing test case. Needs more investigation

**Review by dcleao — APPROVED** [review]
Needs to be merged because the other PRs that depend on this one got merged.
Open issues affecting the tests are to be addressed in a follow-up PR.
